cgi-bin/request_activation

```diff
@@ -88,27 +88,29 @@ if (!defined($code)) {
     $code = &CleanCode($code);
     my $expected_code = &ActivationCode($address);
     if ($code eq $expected_code) {
+        my $report;
         if (UserActivated($optouts, $address)) {
-            print 'already ', $tx->already_activated;
+            $report = 'already ' + $tx->already_activated;
         } else {
-            my $pats = SetOptOutPatterns($optouts, $address, '+');
-            SaveOptOuts($optouts);
-            my $key = &OptOutKey(&CanonicalizeAddr($address));
-            my $fname = "@CIVSDATADIR@/invites/$key";
-            my @rows = ();
-            my $invites = "";
-            if (open(INVITES, "<", $fname)) {
-                $invites = <INVITES>;
-                my @lines = split /\n/, $invites;
-                for my $line (@lines) {
-                    chomp $line;
-                    (my $url, my $title) = split / /, $line, 2;
-                    push @rows, [$url, $title];
-                }
-                close(INVITES)
+            $report = 'activated ' + $tx->activation_successful;
```

Comment:
string concatenation in Perl is . rather than +

Reply:
Whoops! Yes, thank you. Too much recent Python. :)